.devcontainer/Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM mcr.microsoft.com/devcontainers/python:3.12
+LABEL org.opencontainers.image.authors="https://github.com/nqngo"
+LABEL description="Python 3.12 devcontainer with poetry"
+
+# Configure the dev environment
+RUN apt update && \
+    apt install -y bash-completion && \
+    rm -rf /var/lib/apt/list/*
+# Configure poetry
+USER vscode
+WORKDIR /home/vscode
+RUN curl -sSL https://install.python-poetry.org | python3 - && \
```

Comment:
Would we want to install this via `pipx`? It's also recommened via the Poetry site over the curl method :-/

Reply:
Just test the base docker devcontainer image and pipx is available, we can certainly swap to it. I thought the default recommendation method is still using the curl installation script?

Reply:
Literally the first option

![image](https://github.com/bifrostlab/llm-assistant/assets/4188758/272580e6-5bd3-4a91-8651-aab9392aff17)

Reply:
read the doc? What am I? Some kind of dev? I just google it my man. 🤣

Reply:
https://python-poetry.org/docs/#installing-with-pipx
https://python-poetry.org/docs/#ci-recommendations